tensorflow/core/framework/common_shape_fns.cc

```diff
@@ -1050,22 +1050,44 @@ Status ReductionShape(InferenceContext* c) {
   }
 
   const int32 input_rank = c->Rank(input);
-  std::set<int32> true_indices;
-  auto reduction_indices = reduction_indices_t->flat<int32>();
-  for (int i = 0; i < reduction_indices_t->NumElements(); ++i) {
-    int32 reduction_index = reduction_indices(i);
-    if (reduction_index < -input_rank || reduction_index >= input_rank) {
-      return errors::InvalidArgument("Invalid reduction dimension ",
-                                     reduction_index, " for input with ",
-                                     input_rank, " dimensions.");
-    }
+  std::set<int64> true_indices;
+  if (reduction_indices_t->dtype() == DataType::DT_INT32) {
+    auto reduction_indices = reduction_indices_t->flat<int32>();
```

Comment:
Can we pull this out into a little helper function templated on the index type and call it like:

if (...type == DT_INT32) {
  helper_function<int32>(...);
} else if (...type == DT_INT64) {
  helper_function<int64>(...);
} else {
  invalid...
}

Reply:
Thanks @vrv. The PR has been updated with a template helper function of `ReductionShapeHelper`. Please take a look.